runtime/src/main/cpp/Worker.cpp

```diff
@@ -721,10 +759,14 @@ KLong Worker::checkDelayedLocked() {
   }
 }
 
-void Worker::waitForQueueLocked() {
+bool Worker::waitForQueueLocked(KLong timeoutMicroseconds) {
   while (queue_.size() == 0) {
     KLong closestToRun = checkDelayedLocked();
-    if (closestToRun == 0) continue;
+    if (closestToRun == 0) {
+        continue;
+    }
+    if (timeoutMicroseconds >= 0)
+        closestToRun = timeoutMicroseconds < closestToRun || closestToRun < 0 ? timeoutMicroseconds : closestToRun;
```

Comment:
```kotlin
import kotlin.native.concurrent.*

fun main() {
    println(Worker.current.park(2000))
}
```

Finishes much faster than 2000 ms. Is this expected?

Btw, how spurious wakeups are handled here?

Reply:
Argument is microseconds, as explicitly stated in the name.

Reply:
Spurious wakeups are not handled, ant not sure which way/if we shall handle them. If we woke up earlier in timed wait, we'll just check appropriate conditions on that moment. The only questionable part is unlimited wait, but it's also doesn't seem critical.

Reply:
```nike@~/kotlin/kotlin-native
>cat park.kt 
import kotlin.native.concurrent.*

fun main() {
    println(Worker.current.park(2000000))
}
nike@~/kotlin/kotlin-native
>time ./p.kexe
false

real	0m2.013s
user	0m0.005s
sys	0m0.004s
```

Reply:
With new version with timeout in nanoseconds  it has to be 2_000_000_000, of course.

Reply:
> Argument is microseconds, as explicitly stated in the name.
 
Sorry, my bad.

Reply:
> Spurious wakeups are not handled, and not sure which way/if we shall handle them.

For example, by waiting in a loop till a particular moment in time (preferably using monotonic clock if available).

Reply:
Not sure if it is indeed required/important for this functionality.

Reply:
Can current implementation wait longer than specified if spurious wakeups happen?